### PR TITLE
[Enhancement] Refactor Starter Species to use separate EggTier map

### PR DESCRIPTION
**Pull request by ben-lear**
## What are the changes the user will see?

No functional changes.

## Why am I making these changes?
Creates a new Species `EggTier` map that can be used by Balance team to set `EggTier` separately from a species' starter cost. Allows for situations like Regice, where it is still obtainable from Epic eggs as a sub-legendary but has starter cost of 5, etc.

## What are the changes from a developer perspective?
Not much changes, except an egg calculates its `EggTier` from the `speciesEggTier` map now instead of from `speciesStarterCosts`.

Also removed a redundant math clamping function in `utils.ts`

## How to test the changes?
Eggs are hatching with normal expected species. There are no balance changes in this PR.

## Checklist
- [X] **I'm using `beta` as my base branch**
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I considered writing automated tests for the issue?
- [X] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by damocleas — APPROVED** [review]

File: src/utils.ts [hunk 1]
```diff
@@ -38,7 +38,13 @@ export function shiftCharCodes(str: string, shiftCount: integer) {
...
 
-export function clampInt(value: integer, min: integer, max: integer): integer {
+/**
+ * Clamps a number
+ * @param value
+ * @param min
+ * @param max
+ */
+export function clampNumber(value: number, min: number, max: number): number {
   return Math.min(Math.max(value, min), max);
```
Comment (DayKev) [inline src/utils.ts:48]: Tbh this function is unnecessary, there's already `Phaser.Math.Clamp(value, min, max)`, just use that.
```suggestion
  return Phaser.Math.Clamp(value, min, max);
```

File: src/utils.ts [hunk 2]
```diff
@@ -38,7 +38,13 @@ export function shiftCharCodes(str: string, shiftCount: integer) {
   return newStr;
 }
 
-export function clampInt(value: integer, min: integer, max: integer): integer {
+/**
+ * Clamps a number
+ * @param value
+ * @param min
+ * @param max
+ */
```
Comment (DayKev) [inline src/utils.ts:46]:
```suggestion
```

File: src/data/balance/species-egg-tiers.ts [hunk 3]
```diff
@@ -0,0 +1,603 @@
+import { Species } from "#enums/species";
+import { EggTier } from "#enums/egg-type";
+
+/**
+ * Map of all starters and their respective {@link EggTier}, which determines which type of egg the starter hatches from.
```
Comment (DayKev) [inline src/data/balance/species-egg-tiers.ts:5]:
```suggestion
 * Map of all starters and their respective {@linkcode EggTier}, which determines which type of egg the starter hatches from.
```

File: src/utils.ts (hunk 1 above)
Comment (ben-lear) [reply src/utils.ts:48]: we should just remove entirely then

**Review by DayKev — APPROVED** [review]

**Review by torranx — APPROVED** [review]
lgtm

**Review by damocleas — APPROVED** [review]

**Pull request merged**